node_modules/oae-authentication/lib/api.js

```diff
@@ -674,6 +683,118 @@ var getUserIdFromLoginId = module.exports.getUserIdFromLoginId = function(tenant
     _getUserIdFromLoginId(loginId, callback);
 };
 
+
+/**
+ * Get the secret for an existing user's resetpassword request
+ *
+ * @param  {Context}    ctx               Standard context object containing the current user and the current tenant
+ * @param  {String}     userName		  The user's own unique username 
+ * @param  {Function}   callback          Standard callback function
+ * @param  {Object}     callback.err      An error that occurred, if any
+ */
+var getResetPasswordSecret =module.exports.getResetPasswordSecret = function(ctx, tenantAlias, username, callback){
+	
+	// associate the userID with loginID for later operations
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+
+    // Create the loginid object first, so it can be passed into the validation
+    var loginId = new LoginId(tenantAlias, AuthenticationConstants.providers.LOCAL, username);
+	
+	// generate a secret token
+	var secret = crypto.randomBytes(16).toString('hex');
```

Comment:
Please use 4 spaces instead of a tab.

Minor: We have a convention to start comments with a capital (but no trailing dot)